File: src/main/webapp/js/services/marcaService.js
app.factory("marcaService", function ($http) {

	var _carregarMarcas = function () {
		return $http.get("/marca");
	};

	var _salvarMarca = function (marca) {
		return $http.post("/marca", marca);
	};

	return {
		carregarMarcas : _carregarMarcas,
		salvarMarca : _salvarMarca
	};


});

File: src/main/java/br/com/pedidos/services/MarcaService.java
package br.com.pedidos.services;

import java.util.Collections;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import br.com.pedidos.model.Marca;
import br.com.pedidos.repository.MarcaRepository;

@Service
public class MarcaService {
	
	@Autowired
	MarcaRepository marcaRepository;
	
	public Marca salvarLoja (Marca marca) {
		marcaRepository.save(marca);
		
		return marca;
	}
	
	public List<Marca> buscarTodas () {
		
		List<Marca> marcas = marcaRepository.findAll();
		Collections.sort(marcas);
		
		return marcas;
	}

}
